src/settings/JobProfiles/JobProfileDetailsRoute/JobProfileDetailsRoute.js

```diff
@@ -1,32 +1,40 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
 import {
   get,
   find,
 } from 'lodash';
 
 import { JobProfileDetails } from '../JobProfileDetails';
+import { DEFAULT_JOB_PROFILE_ID } from '../../../utils';
 
 const JobProfileDetailsRoute = ({
   resources: {
     jobProfile,
     mappingProfile,
+    jobExecutions,
   },
   history,
   location,
   match,
 }) => {
-  // `find` is used to make sure the matched job profile and mapping profile are displayed to avoid
+  // `find` is used to make sure the matched job profile, mapping profile and job executions are displayed to avoid
   // the flickering because of the disappearing of the previous and appearing of the new ones
   const jobProfileRecord = find([get(jobProfile, 'records.0', {})], { id: match.params.id });
   const mappingProfileRecord = find([get(mappingProfile, 'records.0', {})], { id: jobProfileRecord?.mappingProfileId });
+  const isDefaultProfile = jobProfileRecord?.id === DEFAULT_JOB_PROFILE_ID;
+  const isProfileAlreadyInUse = isDefaultProfile || Boolean(find([get(jobExecutions, 'records.0', {})], { jobProfileId: match.params.id }));
```

Comment:
As far as I understand, we will have to add or extend the same trick in all similar cases. So I think we need a spike to investigate this issue and fix it in the future.

Reply:
As far as my investigation about this point went, this is indeed currently the way to handle this without extra requests. With `resourceShouldRefresh: true` the built-in approach now in the `stripes-connect` is in case some resource in the entire manifest has changed (e.g. condition in the `path` function), then update of all resources in the manifests is triggered. I mean that this is not data-export issue but the approach which is used in the `stripes-connect`. If you think that further investigation is needed, please create the corresponding spike.

Reply:
Agreed to try `useManifest` hook once it is ready. If it won't suit then the spike will be created.